app/lib/katello/util/path_with_substitutions.rb

```diff
@@ -12,12 +14,28 @@ def initialize(path, substitutions)
         @substitutions = substitutions
         @path = path
         @resolved = []
+        generate_substitutions_from_path if @substitutions.empty?
+      end
+
+      def split_path
+        @split ||= @path.split('/').reject(&:blank?)
+      end
+
+      def generate_substitutions_from_path
+        # in RHEL8, arch is typically in the 5th position of the path, so we check here first
+        if split_path.length >= 5 && ARCHITECTURES.include?(split_path[4])
+          @substitutions["basearch"] = split_path[4]
```

Comment:
I'm not sure i understand the value this adds, and i'm not sure i agree that it was agreed upon as being the '5th' directory in the path for rhel8, as i've already found a repo that conflicts with that rule.

Reply:
@jlsherrill would the repo you found (or exceptions you are thinking of) not be covered by the else block? I think I covered this and I even added a test with a repo path that didn't have the arch in the fifth postion.

I discussed the reasoning a bit here https://github.com/Katello/katello/pull/7882#discussion_r241104554 but the basic thought is let's look where RHEL8 is most likely to put the arch first and then check the whole path if its not there.

Reply:
It would, i just don't understand the value that the first check provides over the 2nd check.  The only condition the first check would catch is if there happens to be another arch in the first 4 portions of the path, but we want this particular one?  That seems extremely unlikely to me and a very specific scenario.

Reply:
@jlsherrill I think its worthy to check the expected place for the arch first if we are hearing that they CDN team is trying to place the arch there for RHEL8 repos, even if it is not a guarantee. I understand your point that it would be a unique edge case to have an arch come before, but I also don't see the harm in being a bit more specific about where to check first? I'm just going off what was shared with us

Reply:
I just don't see any value in it?  Especially if RCM isn't even honoring it already.

If i was working in this code, the first thing i'd do is look at this branch condition and wonder why we didn't just use the else statement, since that seems to cover this condition.  Then i'd read about the rhel8 statement and worry that if i touch it i might break something, but never really know what or why, which brings me back to the fact i don't see any value in it (unless I am missing something?)

Reply:
I'm not sure we are going to come to an agreement :) I think we should try to look where the RCM team is making an effort to put the arch first, even if there have been a small number of cases where they haven't put the arch there, the overwhelming majority of RHEL8 repos have it there and it been stated that the arch intended to go in a consistent location.

I think we need some outside input - @beav and/or @jturel do you mind weighing in with your opinion(s)?

Reply:
yep, sounds good :)

Reply:
Joining late but I think I'm following and I understand the question here to be: 'Do we need the 5th position checking for arch in this method' and my response is:

If the tests pass right now (given both branches of `generate_subs_from_path`) and continue to pass without the aforementioned check then we don't need to look at 5th position explicitly. If test fails and it's a valid behavior, then we need to have the check the way it's written or some other way

Reply:
My 2 cents with limited knowledge.. Unless I am missing something the "else" condition handles RHEL8 anyway right? Don't see the need for a make a specific check. If you are worried about paths having 2 arches in it that's probably an RCM issue.